Makefile

```diff
@@ -54,9 +54,13 @@ build: protos build-java build-docker
 # formerly install-python-ci-dependencies-uv-venv
 # editable install
 install-python-dependencies-dev:
-	uv pip sync sdk/python/requirements/py$(PYTHON_VERSION)-ci-requirements.txt
+	uv pip sync --require-hashes sdk/python/requirements/py$(PYTHON_VERSION)-ci-requirements.txt
 	uv pip install --no-deps -e .
 
+install-python-dependencies-limited:
```

Comment:
Could rephrase `limited` to `minimal`? 

The `minimal` is a common term used in software installation.